# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1483530133 18000
#      Wed Jan 04 06:42:13 2017 -0500
# Node ID 7d5c56e38801b14ac4d14b492ee45fe3ea4d7381
# Parent  53bb781d70c0f531eb9dcc3cf0f74e6457054b69
report BLAS integer size in less alarming way

* configure.ac: In configuration summary, always display whether the
detected BLAS integer size is 64 bits instead of only warning if it is
not 64 bits when 64-bit indexing is used.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1022,28 +1022,27 @@ if test $ax_blas_ok = no; then
   esac
 fi
 
 if test $ax_blas_ok = no || test $ax_lapack_ok = no; then
   AC_MSG_ERROR([BLAS and LAPACK libraries are required])
 fi
 
 case $ax_blas_integer_size in
-  4 | 8)
+  4)
+    HAVE_64_BIT_BLAS=no
+  ;;
+  8)
+    HAVE_64_BIT_BLAS=yes
   ;;
   *)
     AC_MSG_ERROR([unrecognized BLAS library integer size])
   ;;
 esac
 
-if test $ENABLE_64 = yes && test $ax_blas_integer_size -ne 8; then
-  warn_blas_integer_size="Your BLAS library doesn't seem to use 64-bit integers.  Some matrix operations will be limited to arrays with dimensions smaller than 2^31 elements."
-  OCTAVE_CONFIGURE_WARNING([warn_blas_integer_size])
-fi
-
 OCTAVE_CHECK_SIZEOF_FORTRAN_INTEGER
 if test $octave_cv_sizeof_fortran_integer -ne $ax_blas_integer_size; then
   if test $ax_blas_integer_size -eq 8; then
     case $F77 in
       *gfortran*)
         case $F77_INTEGER_8_FLAG in
           *-fdefault-integer-8*)
           ;;
@@ -3360,29 +3359,30 @@ Octave is now configured for $canonical_
   X11 libraries:                 $X11_LIBS
   Z CPPFLAGS:                    $Z_CPPFLAGS
   Z LDFLAGS:                     $Z_LDFLAGS
   Z libraries:                   $Z_LIBS
 
   Default pager:                 $DEFAULT_PAGER
   gnuplot:                       $GNUPLOT
 
-  Build Octave Qt GUI:                $build_qt_gui
-  JIT compiler for loops:             $ENABLE_JIT
-  Build Java interface:               $build_java
-  Do internal array bounds checking:  $ENABLE_BOUNDS_CHECK
-  Build static libraries:             $STATIC_LIBS
-  Build shared libraries:             $SHARED_LIBS
-  Dynamic Linking:                    $ENABLE_DYNAMIC_LINKING $DL_API_MSG
-  Include support for GNU readline:   $USE_READLINE
-  64-bit array dims and indexing:     $ENABLE_64
-  OpenMP SMP multithreading:          $ENABLE_OPENMP
-  Build cross tools:                  $cross_tools
-  Build docs:                         $ENABLE_DOCS
-  Truncate intermediate FP results:   $ENABLE_FLOAT_TRUNCATE
+  Build Octave Qt GUI:                  $build_qt_gui
+  JIT compiler for loops:               $ENABLE_JIT
+  Build Java interface:                 $build_java
+  Do internal array bounds checking:    $ENABLE_BOUNDS_CHECK
+  Build static libraries:               $STATIC_LIBS
+  Build shared libraries:               $SHARED_LIBS
+  Dynamic Linking:                      $ENABLE_DYNAMIC_LINKING $DL_API_MSG
+  Include support for GNU readline:     $USE_READLINE
+  64-bit array dims and indexing:       $ENABLE_64
+  64-bit BLAS array dims and indexing:  $HAVE_64_BIT_BLAS
+  OpenMP SMP multithreading:            $ENABLE_OPENMP
+  Build cross tools:                    $cross_tools
+  Build docs:                           $ENABLE_DOCS
+  Truncate intermediate FP results:     $ENABLE_FLOAT_TRUNCATE
 ])
 
 warn_msg_printed=false
 
 OCTAVE_CONFIGURE_WARNING_SUMMARY
 
 if test $ENABLE_DYNAMIC_LINKING = yes; then
   if test $SHARED_LIBS = no; then
